src/main/java/appeng/parts/CableBusContainer.java

```diff
@@ -877,34 +877,36 @@ public void readFromNBT(final CompoundNBT data) {
         this.getFacadeContainer().readFromNBT(data);
     }
 
-    public List<ItemStack> getDrops(final List<ItemStack> drops) {
+    /**
+     * Append the contents of the parts to the drop list, e.g. crafting patterns but not the parts or facades
+     * themselves.
+     */
+    public void appendPartContentDrops(final List<ItemStack> drops) {
         for (final AEPartLocation s : AEPartLocation.values()) {
             final IPart part = this.getPart(s);
             if (part != null) {
-                drops.add(part.getItemStack(PartItemStack.BREAK));
                 part.getDrops(drops, false);
             }
-
-            if (s != AEPartLocation.INTERNAL) {
-                final IFacadePart fp = this.getFacadeContainer().getFacade(s);
-                if (fp != null) {
-                    drops.add(fp.getItemStack());
-                }
-            }
         }
-
-        return drops;
     }
 
-    public List<ItemStack> getNoDrops(final List<ItemStack> drops) {
+    /**
+     * Append the parts and facades to the drop list, but not their contents.
+     */
+    public void appendPartStacks(final List<ItemStack> drops) {
         for (final AEPartLocation s : AEPartLocation.values()) {
             final IPart part = this.getPart(s);
             if (part != null) {
-                part.getDrops(drops, false);
+                drops.add(part.getItemStack(PartItemStack.BREAK));
```

Comment:
should this do an isEmpty() check on the returned stack in case an addon goes crazy?

Reply:
We already use this in various other places like serialisation, so it would already cause issues. Also the helper method itself already ignores any empty itemstack. The whole thing really needs to be redesigned to delegate it fully to the part.